src/main/java/hw_14/mapper/AnimalDeserializer.java

```diff
@@ -0,0 +1,45 @@
+package hw_14.mapper;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import ru.mts.animalstarter.hw_14.entity.Animal;
+import ru.mts.animalstarter.hw_14.entity.Cat;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Base64;
+
+public class AnimalDeserializer  extends StdDeserializer<Cat> {
```

Comment:
// StdKeySerializer is deprecated
ObjectMapper mapper = new ObjectMapper();
mapper.setKeySerializer(new StdKeySerializer());

// New code using JsonSerializer
ObjectMapper mapper = new ObjectMapper();
mapper.setKeySerializer(new JsonSerializer());